stdlib/2/os/__init__.pyi

```diff
@@ -11,98 +15,220 @@ from mypy_extensions import NoReturn
+sysconf_names: Dict[str, int]  # Unix only
+
+EX_OK: int        # Unix only
+EX_USAGE: int     # Unix only
+EX_DATAERR: int   # Unix only
+EX_NOINPUT: int   # Unix only
+EX_NOUSER: int    # Unix only
+EX_NOHOST: int    # Unix only
+EX_UNAVAILABLE: int  # Unix only
+EX_SOFTWARE: int  # Unix only
+EX_OSERR: int     # Unix only
+EX_OSFILE: int    # Unix only
+EX_CANTCREAT: int  # Unix only
+EX_IOERR: int     # Unix only
+EX_TEMPFAIL: int  # Unix only
+EX_PROTOCOL: int  # Unix only
+EX_NOPERM: int    # Unix only
+EX_CONFIG: int    # Unix only
+EX_NOTFOUND: int  # Unix only
+
+P_NOWAIT: int
+P_NOWAITO: int
+P_WAIT: int
+if sys.platform == 'win32':
+    P_DETACH: int  # Windows only
+    P_OVERLAY: int  # Windows only
 
 # wait()/waitpid() options
-WNOHANG = 0  # Unix only
-WCONTINUED = 0  # some Unix systems
-WUNTRACED = 0  # Unix only
-
-TMP_MAX = 0  # Undocumented, but used by tempfile
-_PathType = Union[bytes, Text]
-_StatVFS = NamedTuple('_StatVFS', [('f_bsize', int), ('f_frsize', int), ('f_blocks', int),
-                                   ('f_bfree', int), ('f_bavail', int), ('f_files', int),
-                                   ('f_ffree', int), ('f_favail', int), ('f_flag', int),
-                                   ('f_namemax', int)])
+WNOHANG: int  # Unix only
+WCONTINUED: int  # some Unix systems
+WUNTRACED: int  # Unix only
+
+TMP_MAX: int  # Undocumented, but used by tempfile
+
+# ----- os classes (structures) -----
+if sys.version_info >= (3, 6):
+    from builtins import _PathLike as PathLike  # See comment in builtins
+
+_PathType = path._PathType
+
+if sys.version_info >= (3, 6):
+    class DirEntry(PathLike[AnyStr]):
+        # This is what the scandir interator yields
+        # The constructor is hidden
+
+        name: AnyStr
+        path: AnyStr
+        def inode(self) -> int: ...
+        def is_dir(self, follow_symlinks: bool = ...) -> bool: ...
+        def is_file(self, follow_symlinks: bool = ...) -> bool: ...
+        def is_symlink(self) -> bool: ...
+        def stat(self) -> stat_result: ...
+
+        def __fspath__(self) -> AnyStr: ...
+elif sys.version_info >= (3, 5):
+    class DirEntry(Generic[AnyStr]):
+        # This is what the scandir interator yields
+        # The constructor is hidden
+
+        name: AnyStr
+        path: AnyStr
+        def inode(self) -> int: ...
+        def is_dir(self, follow_symlinks: bool = ...) -> bool: ...
+        def is_file(self, follow_symlinks: bool = ...) -> bool: ...
+        def is_symlink(self) -> bool: ...
+        def stat(self) -> stat_result: ...
+
+
+class stat_result:
+    # For backward compatibility, the return value of stat() is also
+    # accessible as a tuple of at least 10 integers giving the most important
+    # (and portable) members of the stat structure, in the order st_mode,
+    # st_ino, st_dev, st_nlink, st_uid, st_gid, st_size, st_atime, st_mtime,
+    # st_ctime. More items may be added at the end by some implementations.
```

Comment:
Really? That would make classic tuple unpacking impossible. Where did you read that?